### PR TITLE
Add CLI and library to run message classification

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

We use OpenAI functions to run message classification of users' messages, and write the result in a new table called `UserMessageClassification`.

## Deploy Plan

- Deploy on front-edge
- npm run initdb
- run `front> ./admin/cli.sh workspace classify-messages --wId <wId> --limit 10`
-

### CODE REVIEW

File: front/admin/cli.ts [hunk 1]
```diff
@@ -271,6 +272,17 @@ const workspace = async (command: string, args: parseArgs.ParsedArgs) => {
       return;
     }
 
+    case "classify-messages": {
```
Comment (flvndvd) [inline front/admin/cli.ts:275]: Does it need to be in the cli? Shouldn't it be a standalone script?

File: front/admin/tools/message_classification.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    throw new Error("DUST_MANAGED_OPENAI_API_KEY is not set");
+  }
+  const openai = new OpenAI({
+    apiKey: process.env.DUST_MANAGED_OPENAI_API_KEY,
+  });
+
+  const prompt =
+    `Classify this message as one class of the following classes: ${MESSAGE_CLASSES.join(
+      ", "
+    )}:` as const;
```
Comment (flvndvd) [inline front/admin/tools/message_classification.ts:19]: Why do you need the const here?

File: front/admin/tools/message_classification.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      where: {
+        conversationId: conversation.id,
+      },
+      attributes: ["id", "userMessageId"],
+    });
+    for (const message of messages) {
+      if (message.userMessageId) {
+        const userMessage = await UserMessage.findByPk(message.userMessageId);
+        if (userMessage) {
+          console.log(
```
Comment (flvndvd) [inline front/admin/tools/message_classification.ts:87]: Can we use the logger?

File: types/src/shared/message_classification.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  "customer support",
+  "product",
+  "product marketing",
+  "legal",
+  "finance",
+  "hr",
+  "hiring",
+  "ops",
+  "security compliance",
+  "undetermined",
```
Comment (flvndvd) [inline types/src/shared/message_classification.ts:17]: Should we go with "unknown" or "other"?

File: front/admin/tools/message_classification.ts (hunk 3 above)
Comment (lasryaric) [reply front/admin/tools/message_classification.ts:87]: Actually I think console.log are fine here. It's only to be printed in the console of the person who starts the script.

File: front/admin/cli.ts (hunk 1 above)
Comment (lasryaric) [reply front/admin/cli.ts:275]: added a CLI using makeScript() 👍

Comment (flvndvd) [reply front/admin/cli.ts]: Should we remove it from here then :p?

File: front/admin/cli.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { Storage } from "@google-cloud/storage";
 import parseArgs from "minimist";
 import readline from "readline";
 
+import { classifyWorkspace } from "@app/admin/tools/message_classification";
```
Comment (flvndvd) [inline front/admin/cli.ts:7]: ✂️

**Review by flvndvd — APPROVED** [review]

**Pull request merged**